WF_Proton_Launcher/Main.cpp

```diff
@@ -147,6 +150,44 @@ int main(int argc, char** argv)
 	Hashlist hashes;
 	std::vector<std::string> diff;
 
+	if (firstrun) {
+		std::cout << std::endl << "Installing Direct X...";
+
+		FILE* fp;
+		fopen_s(&fp, "directx_Jun2010_redist.exe", "wb");
+		if (!fp)
+		{
+			std::wcout << "Failed to create file on disk." << std::endl;
+		}
+		CURL *curl = curl_easy_init();
+
+		FileWrapper wrap{ fp, {} };
+		curl_easy_setopt(curl, CURLOPT_URL, ("https://download.microsoft.com/download/8/4/A/84A35BF1-DAFE-4AE8-82AF-AD2AE20B6B14/directx_Jun2010_redist.exe"));
+		curl_easy_setopt(curl, CURLOPT_USERAGENT, "Mozilla/5.0 (Warframe_on_Proton)");
+		curl_easy_setopt(curl, CURLOPT_WRITEDATA, &wrap);
+		curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, writeFile);
+
+		CURLcode res;
+		res = curl_easy_perform(curl);
+
+		if (res != CURLE_OK) {
+			std::wcout << "Failed to download Direct X." << std::endl;
+			return 1;
+		}
+
+		curl_easy_cleanup(curl);
+		fclose(fp);
+
+		launch("directx_Jun2010_redist.exe /Q /T:C:\\dx9temp");
+		launch("C:\\dx9temp/DXSETUP.exe /silent");
```

Comment:
Could you remove the redist and the unpacked folder afterwards?

No need to take up unnecessary space after all.

Reply:
aff, forgot to remove the redist. sec.

Reply:
Was just about to mention that, beat me to it :slightly_smiling_face:

Reply:
i also just realize i set the remove as a variable for no reason what so ever.
was checking stackoverflow for the quickest way to do it. copypasta is a learning tool I suppose.